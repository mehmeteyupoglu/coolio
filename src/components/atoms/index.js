import SearchInput from "./search-input.vue";
import CustomSpinner from "./custom-spinner.vue";

export { SearchInput, CustomSpinner };
