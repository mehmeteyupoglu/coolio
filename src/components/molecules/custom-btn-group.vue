<template>
  <v-btn :href="content.path" target="_blank" text>
    <span class="mr-2">{{ content.title }}</span>
    <v-icon>{{ content.icon }}</v-icon>
  </v-btn>
</template>

<script>
/**
 * A @component that produces a button group
 * accepts a content object as props
 * @example <custom-btn-group :content="content"></custom-btn-group>
 */

export default {
  name: "CustomBtnGroup",
  props: {
    /**
     * The content that helps customize btn-group
     * @property: path, title, icon
     */
    content: {
      type: Object,
      default() {
        return {
          path: "/",
          title: "Check out",
          icon: "mdi-open-in-new",
        };
      },
    },
  },

  components: {},

  data: () => ({}),
};
</script>
