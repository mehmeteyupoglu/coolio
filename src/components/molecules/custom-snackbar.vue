<template>
  <v-snackbar bottom dark v-model="content.model" :timeout="content.timeout"
    >{{ content.text }}
    <v-btn
      text
      :href="content.path"
      target="_blank"
      class="subtitle-2 mx-0 px-0 py-0 my-0"
    >
      <strong>{{ content.strong }}</strong></v-btn
    >
  </v-snackbar>
</template>

<script>
/**
 * A @component that produces a snackbar with additional functionality
 * accepts a content object as props
 * @example <custom-snackbar :content="content"></custom-snackbar>
 */

export default {
  name: "CustomSnackbar",
  props: {
    /**
     * The content that helps customize snackbar
     * @property: timeout, model, text, path, strong
     */
    content: {
      type: Object,
      default() {
        return {
          timeout: "-1",
          model: false,
          text: "Don't forget to read",
          path: "/",
          strong: "documentation",
        };
      },
    },
  },

  data: () => ({}),
};
</script>
