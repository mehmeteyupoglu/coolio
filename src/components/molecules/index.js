import CustomBtnGroup from "./custom-btn-group.vue";
import LogoGroup from "./logo-group.vue";
import CustomSnackbar from "./custom-snackbar.vue";
import RenderChips from "./render-chips";

export { CustomBtnGroup, LogoGroup, CustomSnackbar, RenderChips };
