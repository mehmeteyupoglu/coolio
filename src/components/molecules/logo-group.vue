<template>
  <div class="d-flex align-center">
    <v-img
      :alt="content.title + ' Logo'"
      class="shrink mr-2"
      contain
      :src="content.src"
      transition="scale-transition"
      width="40"
    />

    <h1>
      {{ content.title }}
    </h1>
  </div>
</template>

<script>
export default {
  name: "LogoGroup",
  props: {
    content: Object,
  },

  components: {},

  data: () => ({
    //
  }),
};
</script>
