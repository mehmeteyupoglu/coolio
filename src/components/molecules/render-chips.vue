<template>
  <div>
    <v-card-title>Tags</v-card-title>
    <v-card-text>
      <v-chip-group active-class="deep-purple accent-4 white--text" column>
        <v-chip v-for="tag in tags.split(', ')" :key="tag" active>{{
          tag
        }}</v-chip>
      </v-chip-group>
    </v-card-text>
  </div>
</template>

<script>
/**
 * A @component that renders tags in a vuetify chips
 * accepts an string of tags
 * @example <render-chips :tags="tags"></render-chips>
 */

export default {
  name: "RenderChips",
  props: {
    tags: {
      type: String,
      default: () => "Toolio is so cool",
    },
  },

  data: () => ({}),
};
</script>
