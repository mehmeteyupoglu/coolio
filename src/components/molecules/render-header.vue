<template>
  <div>
    <v-img height="250" :src="image"></v-img>

    <v-card-title>{{ title }}</v-card-title>
  </div>
</template>

<script>
/**
 * A @component that renders item header with title and image
 * accepts strings as props
 * @example <render-header :image="image" :title="title"></render-header>
 */

export default {
  name: "RenderHeader",
  props: {
    image: {
      type: String,
      default: () =>
        "https://upload.wikimedia.org/wikipedia/commons/a/ac/No_image_available.svg",
    },
    title: {
      type: String,
      default: () => "Toolio",
    },
  },

  data: () => ({}),
};
</script>
