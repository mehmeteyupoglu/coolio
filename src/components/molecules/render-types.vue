<template>
  <v-card-text>
    <v-row align="center" class="mx-0">
      <div class="my-4 subtitle-1"><strong>Type</strong> â€¢ {{ type }}</div>
    </v-row>
  </v-card-text>
</template>

<script>
/**
 * A @component that renders types within a vuetify card text
 * accepts a string as a prop
 * @example <render-types :type="type"></render-types>
 */

export default {
  name: "RenderTypes",
  props: {
    type: {
      type: String,
      default: () => "Lorem ipsum dolor",
    },
  },

  data: () => ({}),
};
</script>
