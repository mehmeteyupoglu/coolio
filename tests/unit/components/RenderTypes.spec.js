import { mount } from "@vue/test-utils";
import { RenderTypes } from "../../../src/components/molecules";

describe("RenderTypes", () => {
  it("should render types with the props given", () => {
    const wrapper = mount(RenderTypes, {
      propsData: {
        type: "Toolio is so cool",
      },
    });
    expect(wrapper.props().type).toBe("Toolio is so cool");
    expect(wrapper.props().type).not.toBeFalsy();
    expect(wrapper.props().type).not.toBe("Poolio");
    expect(wrapper.props().type).not.toBeNull();
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render types with the props given", () => {
    const wrapper = mount(RenderTypes, {
      propsData: {
        type: "",
      },
    });
    expect(wrapper.props().type).toBe("");
    expect(wrapper.props().type).not.toBeTruthy;
    expect(wrapper.props().type).toBeFalsy();
    expect(wrapper.props().type).not.toBe("Toolio");
    expect(wrapper.props().type).not.toBeNull();
    expect(wrapper.element).toMatchSnapshot();
  });
});
