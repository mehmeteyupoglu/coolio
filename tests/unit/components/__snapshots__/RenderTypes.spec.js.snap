// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderTypes should render types with the props given 1`] = `
<v-card-text>
  <v-row
    align="center"
    class="mx-0"
  >
    <div
      class="my-4 subtitle-1"
    >
      <strong>
        Type
      </strong>
       • Toolio is so cool
    </div>
  </v-row>
</v-card-text>
`;

exports[`RenderTypes should render types with the props given 2`] = `
<v-card-text>
  <v-row
    align="center"
    class="mx-0"
  >
    <div
      class="my-4 subtitle-1"
    >
      <strong>
        Type
      </strong>
       • 
    </div>
  </v-row>
</v-card-text>
`;
